x-pack/libbeat/autodiscover/providers/aws/elb/config.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package elb
+
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/autodiscover/template"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// Config for the aws_elb autodiscover provider.
+type Config struct {
+	Type string `config:"type"`
+
+	// Standard autodiscover fields.
+
+	// Hints are currently not supported, but may be implemented in a later release
+	HintsEnabled bool                    `config:"hints.enabled"`
+	Builders     []*common.Config        `config:"builders"`
+	Appenders    []*common.Config        `config:"appenders"`
+	Templates    template.MapperSettings `config:"templates"`
+
+	// Period defines how often to poll the AWS API.
+	Period time.Duration `config:"period" validate:"nonzero,required"`
+
+	// AWS Specific autodiscover fields
+
+	Regions         []string `config:"regions" validate:"required"`
+	AccessKeyID     string   `config:"access_key_id"`
+	SecretAccessKey string   `config:"secret_access_key"`
+	SessionToken    string   `config:"session_token"`
+}
```

Comment:
Maybe we should add support for profile name here as well? https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html so when using `external.LoadDefaultAWSConfig`, users can specify a profile name.